pynecone/.templates/jinja/web/pages/index.js.jinja2

```diff
@@ -11,6 +11,7 @@
 export default function Component() {
   const [{{state_name}}, {{state_name|react_setter}}] = useState({{initial_state|json_dumps}})
   const [{{const.result}}, {{const.result|react_setter}}] = useState({{const.initial_result|json_dumps}})
+  const [notConnected, setNotConnected] = useState(false)
```

Comment:
Instead of capturing a bool here, this state could store the last connection error.

Then

```
connection_error = BaseVar("connectionError", type_="str", is_local=True)
```

And any page could reference this var

```python
pc.cond(
    connection_error != "",
    error_component(connection_error),
)
```